### DISCUSSION TITLE
`READONLY` command issued each time a command is performed against a slave node with `read_from_replicas=True`?

### DISCUSSION THREAD

**Discussion by rtkcthomson** [Grokzen/redis-py-cluster#455] [Q&A] [answered]
We are using AWS ElastiCache Redis configured in cluster mode with master and slave node pairs. Using a `RedisCluster` instance configured with `read_from_replicas=True` resulted in a noticeably higher CPU load on the slave nodes than on the master nodes while connection count, bytes transferred, GET command count and other metrics appeared to be fairly even between the two. In investigating further we believe that https://github.com/Grokzen/redis-py-cluster/blob/58bafe478f0d77d3c7a77a0533d2b11fcd1d1c9b/rediscluster/client.py#L626 is resulting in a `READONLY` command being issued to the slave nodes prior to every read command that is performed. This uneven distribution of CPU load limits the scalability of the Redis cluster. 

Could the `READONLY` command be performed less often or is it essential to be performed for every read command? For example could it be done once for a connection? Alternatively, could it be pipelined with the originally requested command instead of paying the overhead of having two requests to the slave nodes for each read operation?

**Comment by Grokzen** [comment, marked as answer]
There is a open PR #403 for this to reduce this to only be sent once per connection instead of once per command/pipeline but it has yet to be merged.

**Reply by rtkcthomson** [reply]
Thank you Grokzen, I'll watch the PR.
